pigar/parser.py

```diff
@@ -14,6 +14,18 @@
 from .helpers import parse_git_config, trim_suffix
 
 import nbformat
+try:
+    import IPython
+    # untested
+    # only support IPython >= 6.0.0
+    _transformer_manager = IPython.core.inputtransformer2.TransformerManager()
+
+    # future: support IPython LTS 5.0.0 used by Google Colab
+    # _transformer_manager = ...
+
```

Comment:
What does this mean?

Reply:
It means that the current transformer (parser) only supports IPython > 6, and someone needs to add functionality to support IPython 5 LTS that is used by Google Colab.

I don't have IPython 5 LTS to test it and I don't use it anyway. So, Installing all the dependecy from scratch and make virtual environment is too much hassle (furthermore, I can't run your unittest).

Reply:
None of them seems to be an issue to me, also to you.

You can run the tests like that:

https://github.com/damnever/pigar/blob/823a3c2478361d53dba408bea75e1766d253f3c0/makefile#L10-L11

Reply:
Using IPython 7.0:

======================================================================
FAIL: test_py3_requirements (pigar.tests.test_core.ReqsTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "P:\GitHub\pigar\pigar\tests\test_core.py", line 64, in test_py3_requirements
    self.assertListEqual(sorted(guess.keys()), ['foobar'])
AssertionError: Lists differ: ['foobar', 'subbar', 'subfoo'] != ['foobar']

First list contains 2 additional elements.
First extra element 1:
'subbar'

- ['foobar', 'subbar', 'subfoo']
+ ['foobar']

----------------------------------------------------------------------

Reply:
Also, for IPython 5.0 LTS: [transformer](https://ipython.readthedocs.io/en/5.x/api/generated/IPython.core.inputtransformer.html#IPython.core.inputtransformer.InputTransformer)

I haven't figured out how to use it.

Reply:
I asked [stackoverflow, :)](https://stackoverflow.com/questions/66288018/how-to-use-ipython-transformer-for-ipython-5-x-inputtransformer)